src/TitleDetails/PiecesList/PiecesList.js

```diff
@@ -2,23 +2,60 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { FormattedMessage } from 'react-intl';
 
-import { MultiColumnList } from '@folio/stripes/components';
+import {
+  Icon,
+  MultiColumnList,
+  Tooltip,
+} from '@folio/stripes/components';
 import { FolioFormattedDate } from '@folio/stripes-acq-components';
 
+import { PIECE_FORMAT_LABELS } from '../constants';
+
+import styles from './PiecesList.css';
+
 const columnMapping = {
-  title: <FormattedMessage id="ui-receiving.piece.title" />,
+  barcode: <FormattedMessage id="ui-receiving.piece.barcode" />,
   caption: <FormattedMessage id="ui-receiving.piece.caption" />,
   format: <FormattedMessage id="ui-receiving.piece.format" />,
   receiptDate: <FormattedMessage id="ui-receiving.piece.receiptDate" />,
   receivedDate: <FormattedMessage id="ui-receiving.piece.receivedDate" />,
+  request: <FormattedMessage id="ui-receiving.piece.request" />,
   actions: null,
 };
 
-const PiecesList = ({ pieces, title, visibleColumns, renderActions }) => {
+const PiecesList = ({ pieces, items, requests, visibleColumns, renderActions }) => {
   const formatter = {
-    title: () => title,
+    format: piece => (
+      <span>
+        {PIECE_FORMAT_LABELS[piece.format]}
+        {
+          Boolean(piece.comment) && (
+            <Tooltip
+              id={`piece-comment-${piece.id}`}
+              text={piece.comment}
+            >
+              {({ ref, ariaIds }) => (
+                <Icon
+                  ref={ref}
+                  size="small"
+                  icon="comment"
+                  ariaLabel={ariaIds.text}
+                  iconClassName={styles.pieceComment}
+                />
+              )}
+            </Tooltip>
+          )
+        }
+      </span>
+    ),
     receiptDate: piece => <FolioFormattedDate value={piece.receiptDate} />,
     receivedDate: piece => <FolioFormattedDate value={piece.receivedDate} />,
+    barcode: piece => (items.find(item => item.id === piece.itemId) || {}).barcode,
+    request: piece => (
+      requests.find(request => request.itemId === piece.itemId)
+        ? <FormattedMessage id="ui-receiving.piece.request.isOpened" />
+        : ''
```

Comment:
it's a dash in one mock-up, maybe it should be consistent https://issues.folio.org/browse/UIREC-53
https://issues.folio.org/secure/attachment/24840/2-receive-multiples.png

Reply:
on receive modal it's not dash and previously it was required to display nothing